### PR TITLE
[MEX-414] composable tasks integration

### PR DESCRIPTION
**Pull request by claudiulataretu**
## Reasoning
- composable tasks contract can execute in a single transaction multiple tasks
  
## Proposed Changes
- added new module for composable task contract integration
- updated swap fixed input transactions from pair to use wrap or unwrap and swap fixed input tasks
- updated multi pair swap with fixed input transactions to use wrap or unwrap and multi swap tasks

## How to test
```
query Swap {
	swap(
		tokenInID: "EGLD",
		tokenOutID: <token_id>,
		tolerance: 0.01,
		amountIn: "1000000000000000000"
	) {
		swapType
		amountIn
		amountOut
		fees
		pricesImpact
		tokenInExchangeRate
		tokenOutExchangeRate
		transactions {
			receiver
			value
			data
		}
	}
}
```
- query should return a single transaction
```
query SwapTokensFixedInput {
	swapTokensFixedInput(
		pairAddress: <pair_address>,
		tokenInID: "EGLD",
		tokenOutID: <token_id>,
		amountIn: "1000000000000000000",
		amountOut: <amount_out>,
		tolerance: 0.01
	) {
		data
	}
}
```
- query should return a single transaction

### CODE REVIEW

File: src/modules/composable-tasks/services/composable.tasks.transaction.ts [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    arguments: BytesValue[];
+};
+
+@Injectable()
+export class ComposableTasksTransactionService {
+    constructor(
+        private readonly mxPorxy: MXProxyService,
+        private readonly wrapAbi: WrapAbiService,
+    ) {}
+
```
Comment (danielailie) [inline src/modules/composable-tasks/services/composable.tasks.transaction.ts:41]: I would split this method into multiple smaller methods. for example getRawTasks and buildTransaction

**Review by danielailie — APPROVED** [review]

File: src/modules/composable-tasks/services/composable.tasks.transaction.ts (hunk 1 above)
Comment (claudiulataretu) [reply src/modules/composable-tasks/services/composable.tasks.transaction.ts:41]: done

**Review by danielailie — DISMISSED** [review]

File: src/modules/composable-tasks/services/composable.tasks.transaction.ts [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    async getComposeTasksTransaction(
+        payment: EsdtTokenPayment,
+        tokenOut: EgldOrEsdtTokenPayment,
+        tasks: ComposableTask[],
+    ): Promise<TransactionModel> {
+        const contract = await this.mxPorxy.getComposableTasksSmartContract();
```
Comment (mad2sm0key) [inline src/modules/composable-tasks/services/composable.tasks.transaction.ts:47]: small typo "mxPorxy"

**Review by mad2sm0key — APPROVED** [review]

File: src/modules/composable-tasks/services/composable.tasks.transaction.ts (hunk 2 above)
Comment (claudiulataretu) [reply src/modules/composable-tasks/services/composable.tasks.transaction.ts]: done

**Review by mad2sm0key — APPROVED** [review]

**Review by danielailie — APPROVED** [review]

**Pull request merged**